### PR TITLE
Reorder Java index documentation

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #453
Reorder Java topics to follow the same order as the Scala ones

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -78,18 +75,18 @@ The steps are exactly the same for everyone involved in the project (be it core
...
-1. Fork Akka
+1. Fork akka-http
 2. Hack and test on your feature (on a branch)
 3. Submit a PR
 4. Sign the CLA if necessary
 4. Keep polishing it until received enough LGTM
 5. Profit!
 
-Note that the Akka sbt project is large, so `sbt` needs to be run with lots of heap (1-2 Gb). This can be specified using a command line argument `sbt -mem 2048` or in the environment variable `SBT_OPTS` but then as a regular JVM memory flag, for example `SBT_OPTS=-Xmx2G`, on some platforms you can also edit the global defaults for sbt in `/usr/local/etc/sbtopts`.
+Note that the akka-http sbt project is not as large as the Akka one, so `sbt` should be able to run with less heap than with the Akka project. In case you need to increase the heap, this can be specified using a command line argument `sbt -mem 2048` or in the environment variable `SBT_OPTS` but then as a regular JVM memory flag, for example `SBT_OPTS=-Xmx2G`, on some platforms you can also edit the global defaults for sbt in `/usr/local/etc/sbtopts`.
```
Comment (2beaucoup) [inline CONTRIBUTING.md]: 😄

**Comment by jlprat** [comment]
Sure, give me some minutes!

**Comment by 2beaucoup** [comment]
There's no rush. (Looks like it's actually just the last commit that adds on top of master BTW.)

**Comment by jrudolph** [comment]
If it's just one commit in the first place we can just squash it here.

**Comment by jlprat** [comment]
One of the last PR's I did I merged it also into my fork's master by mistake

**Comment by jrudolph** [comment]
Yes, no worries. Squashed and merged now as a single commit. Thanks, again :)

**Comment by jlprat** [comment]
Happy to contribute! :+1:

**Comment by 2beaucoup** [comment]
Need to work on my merge foo. Thanks for taking over @jrudolph!

**Comment by ktoso** [comment]
Great, thank you @jlprat !

**Pull request merged**